### PR TITLE
implement cov

### PR DESCRIPTION
**Pull request by Hokchris**
https://github.com/unifyai/ivy/issues/6269

closes https://github.com/unifyai/ivy/issues/6269

Implemented cov for linear algebra section.

### CODE REVIEW

File: ivy/functional/backends/tensorflow/linear_algebra.py [hunk 1]
```diff
@@ -35,6 +35,114 @@ def cholesky(
...
+    if x2 is not None:
+        x2 = tf.experimental.numpy.array(x2, copy=False, ndmin=2, dtype=dtype)
+        if not rowVar and tf.shape(x2)[0] != 1:
+            x2 = tf.transpose(x2)
+
+        X = tf.concat([X, x2], axis=0)
+
+    w = None
+    if fweights is not None:
+        fweights = tf.cast(fweights, dtype=tf.float64)
```
[the remark this answers is not in the document]
Comment (AnnaTz) [reply ivy/functional/backends/tensorflow/linear_algebra.py:92]: I'm not sure casting `fweights` and `aweights` makes any difference in the results. Is there a specific reason you have added it in all backends?

File: ivy_tests/test_ivy/test_functional/test_core/test_linalg.py [hunk 2]
```diff
@@ -1292,6 +1381,53 @@ def test_cholesky(
...
+        max_num_dims=2,
+        min_dim_size=2,
+        max_dim_size=5,
+        min_value=1,
+        max_value=1e10,
+        abs_smallest_val=0.01,
+        large_abs_safety_factor=2,
+        safety_factor_scale="log",
+    ),
+    test_gradients=st.just(False),
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_functional/test_core/test_linalg.py:1399]: If you remove the `out` argument, you should add `test_with_out=st.just(False),` here.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by Hokchris** [comment]
Hey! Just went through the requested changes and I think they should be done now. The one thing I wasn't sure about making changes to were some of the datatype casts, namely in the Jax, Numpy and Tensorflow backends. 

Jax's default datatype is float32 and I get a datatype warning that it doesn't match with the ground truth backend (Tensorflow and Numpy both give the same complaint I think).
The three backends give an error if I get rid of the fweights datatype casting to _int64_ (I removed the aweights and they all still worked). Jax and Numpy both say that fweights must be integer (it should be with the dtype specified in the test file, but it looks like the fweights are being cast to floats when being passed into the backends), and Tensorflow doesn't allow **w** * **aweights** on line 117 without the casting ("cannot compute Mul as input #1(zero-based) was expected to be a double tensor but is a float tensor").
 
Is there another flag that I could use in the test file to prevent the datatype casting? The datatypes for fweights and aweights are currently hardcoded into the dtypes input array.

I assume I should make another PR due to the recent force push again to resolve the conflicts above?

**Comment by AnnaTz** [comment]
> 

Hi!
Since the casting is there to prevent certain errors you were getting, you shouldn't remove them of course. I just wasn't sure what reason you had to add them in the first place. 
As far as I know there is no flag to disable that. That's why we are using the `input_dtypes` argument.
Yes, after the force-push all PRs need to be re-created.

**Comment by AnnaTz** [comment]
> Hey! Just went through the requested changes and I think they should be done now. The one thing I wasn't sure about making changes to were some of the datatype casts, namely in the Jax, Numpy and Tensorflow backends.
> 
> Jax's default datatype is float32 and I get a datatype warning that it doesn't match with the ground truth backend (Tensorflow and Numpy both give the same complaint I think). The three backends give an error if I get rid of the fweights datatype casting to _int64_ (I removed the aweights and they all still worked). Jax and Numpy both say that fweights must be integer (it should be with the dtype specified in the test file, but it looks like the fweights are being cast to floats when being passed into the backends), and Tensorflow doesn't allow **w** * **aweights** on line 117 without the casting ("cannot compute Mul as input #1(zero-based) was expected to be a double tensor but is a float tensor").
> 
> Is there another flag that I could use in the test file to prevent the datatype casting? The datatypes for fweights and aweights are currently hardcoded into the dtypes input array.
> 
> I assume I should make another PR due to the recent force push again to resolve the conflicts above?

Hi!
Since the castings are there to prevent certain errors you were getting, you shouldn't remove them of course. I just wasn't sure what was the reason you had added them.
As far as I know there is no flag to prevent that. That's why we are using the `input_dtypes` argument.
Yes, after the force-push all PRs need to be re-created.

**Comment by Hokchris** [comment]
Hey, I think I might've accidentally closed this yesterday when trying to re-commit the changes to a new branch. Should I contact someone about issue 6269 (https://github.com/unifyai/ivy/issues/6269) being closed by ivy-leaves yesterday (I assume automatically) or can these changes be done via this PR?

**Comment by AnnaTz** [comment]
> Hey, I think I might've accidentally closed this yesterday when trying to re-commit the changes to a new branch. Should I contact someone about issue 6269 (#6269) being closed by ivy-leaves yesterday (I assume automatically) or can these changes be done via this PR?

No worries, I opened the issue again and re-linked it to the ToDo list.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by Hokchris** [comment]
Hey, just made a small change removing a flag from the test. I had an issue trying to run the tests locally after the previous force push and I started getting an assertion error: "ground truth backend (tensorflow) returned array on device gpu:0 but target backend (torch) returned array on device cpu" regardless of the function I was trying to test, but I just set up the environment with a docker container and it worked as intended.

Are there still some changes I should make or is this version fine as the tests look to pass locally at least for me?

**Comment by AnnaTz** [comment]
Hi @Hokchris, unfortunately there's been another force push in our repository so you'll need to create a new PR. Apologies for the inconvenience. 
If the tests are passing locally I guess the new PR will be able to get merged fairly quickly.

**Pull request closed without merging**